src/javaScript/cart.js

```diff
@@ -101,25 +118,63 @@ function updateQuantityColor () {
     return
   }
 
-  addItems.forEach((addItem, index) => {
+  addItems.forEach((addItem) => {
     addItem.addEventListener('click', () => {
-      const quantity = quantities[index]
-      quantity.style.color = 'green'
-      quantity.textContent = parseInt(quantity.textContent) + 1
+      const index = addItem.dataset.index
+      const cartItems = getCartItems()
+      cartItems[index].quantity += 1
+      setCartItems(cartItems)
+      quantities[index].textContent = cartItems[index].quantity
+      quantities[index].style.color = 'green'
+
+      const priceNumber = parseFloat(
+        cartItems[index].price.replace(/[^0-9.-]+/g, '')
+      )
+      const newSubtotal = priceNumber * cartItems[index].quantity
+      const subtotalElement = document.querySelector(
+        `.subtotal[data-index="${index}"]`
+      )
+      subtotalElement.textContent = `$${newSubtotal.toFixed(2)}`
+
       updateButton.removeAttribute('disabled')
+      updateButton.classList.remove('text-muted')
     })
   })
 
-  subtractItems.forEach((subtractItem, index) => {
+  subtractItems.forEach((subtractItem) => {
     subtractItem.addEventListener('click', () => {
-      const quantity = quantities[index]
-      quantity.style.color = 'red'
-      quantity.textContent = Math.max(0, parseInt(quantity.textContent) - 1)
+      const index = subtractItem.dataset.index
+      const cartItems = getCartItems()
+      cartItems[index].quantity = Math.max(0, cartItems[index].quantity - 1)
+      setCartItems(cartItems)
+      quantities[index].textContent = cartItems[index].quantity
+      quantities[index].style.color = 'red'
+
+      const priceNumber = parseFloat(
+        cartItems[index].price.replace(/[^0-9.-]+/g, '')
```

Comment:
Why is this needed?

Reply:
To prevent any error when we convert strings to numbers, so by using `.replace` i ensure that only numeric characters, the decimal point, and the minus sign are kept